pandas/core/reshape/pivot.py

```diff
@@ -424,16 +435,27 @@ def _convert_by(by):
 
 @Substitution("\ndata : DataFrame")
 @Appender(_shared_docs["pivot"], indents=1)
-def pivot(data: "DataFrame", index=None, columns=None, values=None) -> "DataFrame":
+def pivot(
+    data: "DataFrame",
+    index: Optional[Union[Label, Sequence[Label]]] = None,
+    columns: Optional[Union[Label, Sequence[Label]]] = None,
+    values: Optional[Union[Label, Sequence[Label]]] = None,
+) -> "DataFrame":
     if columns is None:
         raise TypeError("pivot() missing 1 required argument: 'columns'")
-    columns = columns if is_list_like(columns) else [columns]
+
+    if is_list_like(columns):
+        columns = cast(Sequence[Label], columns)
+        columns = columns
+    else:
+        columns = [columns]
 
     if values is None:
-        cols: List[str] = []
+        cols: List[Label] = []
         if index is None:
             pass
         elif is_list_like(index):
+            index = cast(Sequence[Label], index)
```

Comment:
same

Reply:
emm, this one might be better to keep as is, since we are not directly converting `index` to list like above, but to use index to get `[data[idx] for idx in index]`.